docs/hoon/demo.md

```diff
@@ -24,65 +25,60 @@ line is commented with its number:
-side effect, but officially Hoon has no side effects.  Accordingly,
-this code will *produce* a list of strings, not *print* a list of
-strings. (Of course, if you run this code from the Urbit
-[_Dojo_](../../using/shell) (shell) you will be able to see the
-product.)
+When FizzBuzz is written in non-functional languages the 
+output is generated as a side effect of the program, e.g., with 
+`printf()`.  Hoon is a functional language so technically it 
+has no side effects.  Accordingly, the code above *returns* a 
+list of strings, it doesn't print them. (If you run it from the 
+Urbit shell, [_Dojo_](../../using/shell), you will of course be 
+able to see the product.)
 
-The code produces a list in which each item is either a number
-(as a string), "Fizz", "Buzz", or "Fizzbuzz". The value `~` is
-"null" and indicates the end of the list.
+The code produces a list in which each item is either a number, 
+"Fizz", "Buzz", or "Fizzbuzz". The value `~` is "null" and indicates 
+the end of the list.
 
-Line 1 uses the _[rune](../../about/glossary#rune)_ `|=` (pronounced
-["bar-tis"](../syntax/#-glyphs-and-characters)) to define a function
-that takes one argument, labelled `end`. The argument must be of the
-type `@ud`, an unsigned decimal number.
+Line 1 uses the _[rune](../../about/glossary#rune)_ `|=` to define 
+a function that takes one input, `end`. The input value must be of 
+the type `@`, an atom (i.e., an unsigned integer).
 
-Line 2 uses the rune `=/` (pronounced "tis-fas") to declare a variable,
-`count` of the unsigned decimal number type, with initial value `1`.
+Line 2 uses `=/` to declare a variable, `count`, as an atom with an 
+initial value of `1`.
 
-Line 3 uses the rune `|-` ("bar-hep") to begin a loop.
+Line 3 uses `|-` to indicate the starting point of a loop 
+(technically a recursion).
 
 Line 4 defines the product of the loop as a list of `tape`s.
 (A `tape` is a list of characters, one of Hoon's two string 
-types.)  The rune `^-` ("ket-hep") is used for defining
-the type of value the function must produce.
+types.)  The rune `^-` is used for defining the type of value the 
+function is required to produce.
 
-Line 5 uses the rune `?:` ("wut-col"), which is your classic "if, then,
-else" conditional. The "if" clause uses the irregular form of the rune
-`.=` ("dot-tis"), one of the fundamental [Nock](../../nock/definition)
-instructions, to perform an equality test on the values of `count` and
-`end`. This will produce either `%.y` (yes, the number 0) or `%.n`
-(no, 1) and consequently compute the "then" or "else" clause
-respectively.
+Line 5 uses the rune `?:` which is an ordinary "if-then-else" 
+conditional.  If `=(end count)` evaluates as true, then line 6 is 
+evaluated next.  Otherwise line 7 and on is evaluated.
 
-If the values of `count` and `end` are equal, line 6 (and the whole loop)
-returns the value `~`. The list is finished.  
+If the values of `count` and `end` are equal then line 6 returns the 
```

Comment:
Here and prior, are we comfortable use "return" and "produce" interchangeably in introductory docs?

Reply:
I am because this particular doc is meant to be more intuitive than rigorous, but I don't see any reason not to switch to "produce" here.